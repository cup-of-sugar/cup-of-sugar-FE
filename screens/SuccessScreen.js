import * as WebBrowser from "expo-web-browser";
import * as React from "react";
import {
  Image,
  Platform,
  StyleSheet,
  Text,
  TouchableOpacity,
  View
} from "react-native";
import Colors from "../constants/Colors";

export default function SuccessScreen(props) {
  const action = props.route.params.action;
  const userId = props.route.params.userId;
  const name = props.route.params.name;
  const quantity = props.route.params.quantity;
  const measurement = props.route.params.measurement;
  const timeDuration = props.route.params.timeDuration;

  return (
    <View style={styles.container}>
      <Text style={styles.successMessage}>
        You have successfully{" "}
        {action === "borrow"
          ? "borrowed"
          : action === "return"
          ? "returned"
          : action === "request"
          ? "requested"
          : action === "lend"
          ? "offered to loan your"
          : ""}
        :
      </Text>
      {action === "lend" || action === "request" ? (
        <Text style={styles.itemInfo}>{name}</Text>
      ) : !timeDuration ? (
        <Text style={styles.itemInfo}>
          {quantity} {measurement + " of"} {name}
        </Text>
      ) : (
        <Text style={styles.itemInfo}>
          1 {name}{" "}
          {action === "borrow" ? `for ${quantity} ${timeDuration}` : ""}
        </Text>
      )}
      {action === "borrow" ? (
        <>
          <Text style={styles.successMessage}>
            Message the lender to coordinate pick-up details!
          </Text>
          <TouchableOpacity
            style={styles.messageButton}
            onPress={() =>
              props.navigation.navigate("Compose", { action, userId })
            }
          >
            <Text style={styles.messageButtonText}>Send Message</Text>
          </TouchableOpacity>
          <TouchableOpacity
            style={styles.messageButton}
            onPress={() =>
              props.navigation.navigate("Home", { action, userId })
            }
          >
            <Text style={styles.messageButtonText}>Message Later</Text>
          </TouchableOpacity>
        </>
      ) : (
        <>
          <Text style={styles.successMessage}>
            The neighbor who requested to borrow this item will reach out to you
            soon. Thank you!
          </Text>
          <TouchableOpacity
            style={styles.messageButton}
            onPress={() =>
              props.navigation.navigate("Home", { action, userId })
            }
          >
            <Text style={styles.messageButtonText}>
              {action === "lend" ? "Loan" : "Borrow"} Another Item
            </Text>
          </TouchableOpacity>
        </>
      )}
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: "#fff",
    paddingTop: 80
  },
  successMessage: {
    fontSize: 22,
    fontWeight: "bold",
    color: "black",
    lineHeight: 35,
    margin: 20,
    textAlign: "center"
  },
  itemInfo: {
    alignSelf: "center",
    backgroundColor: Colors.darkBlue,
    borderRadius: 18,
    borderWidth: 1,
    borderColor: Colors.darkBlue,
    fontSize: 26,
    fontWeight: "bold",
    color: Colors.aqua,
    height: 70,
    lineHeight: 35,
    paddingTop: 15,
    marginBottom: 60,
    overflow: "hidden",
    width: 360,
    textAlign: "center"
  },
  messageButton: {
    margin: 40,
    marginTop: 10,
    paddingTop: 20,
    paddingBottom: 20,
    backgroundColor: Colors.lightBlue,
    borderRadius: 10,
    borderWidth: 1,
    borderColor: "#fff"
  },
  messageButtonText: {
    fontSize: 25,
    fontWeight: "bold",
    color: "#fff",
    textAlign: "center"
  }
});
