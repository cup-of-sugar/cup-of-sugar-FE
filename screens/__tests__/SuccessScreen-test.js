import React from 'react';
import SuccessScreen from '../SuccessScreen';

import renderer from 'react-test-renderer';

test('should render component', () => {
  const props = {
    route: {
      params: {
        action: 'borrow',
        name: 'butter',
        quantity: 1,
        measurement: 'oz',
        timeDuration: null,
      },
    },
  };
  const tree = renderer.create(<SuccessScreen {...props} />).toJSON();
  expect(tree).toMatchSnapshot();
});

test('should render component', () => {
  const props = {
    route: {
      params: {
        action: 'lend',
        name: 'butter',
        quantity: 1,
        measurement: 'oz',
        timeDuration: null,
      },
    },
  };
  const tree = renderer.create(<SuccessScreen {...props} />).toJSON();
  expect(tree).toMatchSnapshot();
});

test('should render component', () => {
  const props = {
    route: {
      params: {
        action: 'request',
        name: 'butter',
        quantity: 1,
        measurement: 'oz',
        timeDuration: null,
      },
    },
  };
  const tree = renderer.create(<SuccessScreen {...props} />).toJSON();
  expect(tree).toMatchSnapshot();
});

test('should render component', () => {
  const props = {
    route: {
      params: {
        action: 'return',
        name: 'butter',
        quantity: 1,
        measurement: 'oz',
        timeDuration: null,
      },
    },
  };
  const tree = renderer.create(<SuccessScreen {...props} />).toJSON();
  expect(tree).toMatchSnapshot();
});

test('should render component', () => {
  const props = {
    route: {
      params: {
        action: 'borrow',
        name: 'drill',
        quantity: 1,
        measurement: null,
        timeDuration: 'day',
      },
    },
  };
  const tree = renderer.create(<SuccessScreen {...props} />).toJSON();
  expect(tree).toMatchSnapshot();
});
