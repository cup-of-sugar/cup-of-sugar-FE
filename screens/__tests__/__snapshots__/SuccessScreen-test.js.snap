// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render component 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#fff",
      "flex": 1,
      "paddingTop": 80,
    }
  }
>
  <Text
    style={
      Object {
        "color": "black",
        "fontSize": 22,
        "fontWeight": "bold",
        "lineHeight": 35,
        "margin": 20,
        "textAlign": "center",
      }
    }
  >
    You have successfully
     
    borrowed
    :
  </Text>
  <Text
    style={
      Object {
        "alignSelf": "center",
        "backgroundColor": "#385A94",
        "borderColor": "#385A94",
        "borderRadius": 18,
        "borderWidth": 1,
        "color": "#0CE1C3",
        "fontSize": 28,
        "fontWeight": "bold",
        "height": 70,
        "lineHeight": 35,
        "marginBottom": 60,
        "paddingTop": 15,
        "textAlign": "center",
        "width": 300,
      }
    }
  >
    1
     
    oz of
     
    butter
  </Text>
  <Text
    style={
      Object {
        "color": "black",
        "fontSize": 22,
        "fontWeight": "bold",
        "lineHeight": 35,
        "margin": 20,
        "textAlign": "center",
      }
    }
  >
    Message the lender to coordinate pick-up details!
  </Text>
  <View
    accessible={true}
    focusable={true}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "#0C94E1",
        "borderColor": "#fff",
        "borderRadius": 10,
        "borderWidth": 1,
        "margin": 40,
        "marginTop": 10,
        "opacity": 1,
        "paddingBottom": 20,
        "paddingTop": 20,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#fff",
          "fontSize": 25,
          "fontWeight": "bold",
          "textAlign": "center",
        }
      }
    >
      Send Message
    </Text>
  </View>
</View>
`;

exports[`should render component 2`] = `
<View
  style={
    Object {
      "backgroundColor": "#fff",
      "flex": 1,
      "paddingTop": 80,
    }
  }
>
  <Text
    style={
      Object {
        "color": "black",
        "fontSize": 22,
        "fontWeight": "bold",
        "lineHeight": 35,
        "margin": 20,
        "textAlign": "center",
      }
    }
  >
    You have successfully
     
    offered to loan your
    :
  </Text>
  <Text
    style={
      Object {
        "alignSelf": "center",
        "backgroundColor": "#385A94",
        "borderColor": "#385A94",
        "borderRadius": 18,
        "borderWidth": 1,
        "color": "#0CE1C3",
        "fontSize": 28,
        "fontWeight": "bold",
        "height": 70,
        "lineHeight": 35,
        "marginBottom": 60,
        "paddingTop": 15,
        "textAlign": "center",
        "width": 300,
      }
    }
  >
    butter
  </Text>
  <Text
    style={
      Object {
        "color": "black",
        "fontSize": 22,
        "fontWeight": "bold",
        "lineHeight": 35,
        "margin": 20,
        "textAlign": "center",
      }
    }
  >
    Thank you!
  </Text>
  <View
    accessible={true}
    focusable={true}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "#0C94E1",
        "borderColor": "#fff",
        "borderRadius": 10,
        "borderWidth": 1,
        "margin": 40,
        "marginTop": 10,
        "opacity": 1,
        "paddingBottom": 20,
        "paddingTop": 20,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#fff",
          "fontSize": 25,
          "fontWeight": "bold",
          "textAlign": "center",
        }
      }
    >
      Loan
       Another Item
    </Text>
  </View>
</View>
`;

exports[`should render component 3`] = `
<View
  style={
    Object {
      "backgroundColor": "#fff",
      "flex": 1,
      "paddingTop": 80,
    }
  }
>
  <Text
    style={
      Object {
        "color": "black",
        "fontSize": 22,
        "fontWeight": "bold",
        "lineHeight": 35,
        "margin": 20,
        "textAlign": "center",
      }
    }
  >
    You have successfully
     
    requested
    :
  </Text>
  <Text
    style={
      Object {
        "alignSelf": "center",
        "backgroundColor": "#385A94",
        "borderColor": "#385A94",
        "borderRadius": 18,
        "borderWidth": 1,
        "color": "#0CE1C3",
        "fontSize": 28,
        "fontWeight": "bold",
        "height": 70,
        "lineHeight": 35,
        "marginBottom": 60,
        "paddingTop": 15,
        "textAlign": "center",
        "width": 300,
      }
    }
  >
    butter
  </Text>
  <Text
    style={
      Object {
        "color": "black",
        "fontSize": 22,
        "fontWeight": "bold",
        "lineHeight": 35,
        "margin": 20,
        "textAlign": "center",
      }
    }
  >
    Thank you!
  </Text>
  <View
    accessible={true}
    focusable={true}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "#0C94E1",
        "borderColor": "#fff",
        "borderRadius": 10,
        "borderWidth": 1,
        "margin": 40,
        "marginTop": 10,
        "opacity": 1,
        "paddingBottom": 20,
        "paddingTop": 20,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#fff",
          "fontSize": 25,
          "fontWeight": "bold",
          "textAlign": "center",
        }
      }
    >
      Borrow
       Another Item
    </Text>
  </View>
</View>
`;

exports[`should render component 4`] = `
<View
  style={
    Object {
      "backgroundColor": "#fff",
      "flex": 1,
      "paddingTop": 80,
    }
  }
>
  <Text
    style={
      Object {
        "color": "black",
        "fontSize": 22,
        "fontWeight": "bold",
        "lineHeight": 35,
        "margin": 20,
        "textAlign": "center",
      }
    }
  >
    You have successfully
     
    returned
    :
  </Text>
  <Text
    style={
      Object {
        "alignSelf": "center",
        "backgroundColor": "#385A94",
        "borderColor": "#385A94",
        "borderRadius": 18,
        "borderWidth": 1,
        "color": "#0CE1C3",
        "fontSize": 28,
        "fontWeight": "bold",
        "height": 70,
        "lineHeight": 35,
        "marginBottom": 60,
        "paddingTop": 15,
        "textAlign": "center",
        "width": 300,
      }
    }
  >
    1
     
    oz of
     
    butter
  </Text>
  <Text
    style={
      Object {
        "color": "black",
        "fontSize": 22,
        "fontWeight": "bold",
        "lineHeight": 35,
        "margin": 20,
        "textAlign": "center",
      }
    }
  >
    Thank you!
  </Text>
  <View
    accessible={true}
    focusable={true}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "#0C94E1",
        "borderColor": "#fff",
        "borderRadius": 10,
        "borderWidth": 1,
        "margin": 40,
        "marginTop": 10,
        "opacity": 1,
        "paddingBottom": 20,
        "paddingTop": 20,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#fff",
          "fontSize": 25,
          "fontWeight": "bold",
          "textAlign": "center",
        }
      }
    >
      Borrow
       Another Item
    </Text>
  </View>
</View>
`;

exports[`should render component 5`] = `
<View
  style={
    Object {
      "backgroundColor": "#fff",
      "flex": 1,
      "paddingTop": 80,
    }
  }
>
  <Text
    style={
      Object {
        "color": "black",
        "fontSize": 22,
        "fontWeight": "bold",
        "lineHeight": 35,
        "margin": 20,
        "textAlign": "center",
      }
    }
  >
    You have successfully
     
    borrowed
    :
  </Text>
  <Text
    style={
      Object {
        "alignSelf": "center",
        "backgroundColor": "#385A94",
        "borderColor": "#385A94",
        "borderRadius": 18,
        "borderWidth": 1,
        "color": "#0CE1C3",
        "fontSize": 28,
        "fontWeight": "bold",
        "height": 70,
        "lineHeight": 35,
        "marginBottom": 60,
        "paddingTop": 15,
        "textAlign": "center",
        "width": 300,
      }
    }
  >
    1 
    drill
     
    for 1 day
  </Text>
  <Text
    style={
      Object {
        "color": "black",
        "fontSize": 22,
        "fontWeight": "bold",
        "lineHeight": 35,
        "margin": 20,
        "textAlign": "center",
      }
    }
  >
    Message the lender to coordinate pick-up details!
  </Text>
  <View
    accessible={true}
    focusable={true}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "#0C94E1",
        "borderColor": "#fff",
        "borderRadius": 10,
        "borderWidth": 1,
        "margin": 40,
        "marginTop": 10,
        "opacity": 1,
        "paddingBottom": 20,
        "paddingTop": 20,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#fff",
          "fontSize": 25,
          "fontWeight": "bold",
          "textAlign": "center",
        }
      }
    >
      Send Message
    </Text>
  </View>
</View>
`;
